core/db/migrate/20210114220232_remove_store_id_from_spree_payment_methods.rb

```diff
@@ -0,0 +1,5 @@
+class RemoveStoreIdFromSpreePaymentMethods < ActiveRecord::Migration[6.0]
+  def change
+    remove_column :spree_payment_methods, :store_id
```

Comment:
@MatthewKennedy we should migrate all of the current settings to the new structure before removing the column :)

Reply:
```ruby
Spree::PaymentMethod.each do |payment_method|
  payment_method.stores_ids = [store_id]
  payment_method.save
end
```

should work :)

Reply:
Thanks for that, I was unsure how to go about the data migration.

Just working on this PR now so will push an update soon.

Reply:
@MatthewKennedy no prob, this code should be run before removing the `store_id` column of course

Reply:
@damianlegawiec How shall we handle nil value in the store_id column? A nil value currently means all stores, so we would have to populate a join for each available store to replicate the previous behaviour.

Is this code in the right direction?
```
  Spree::PaymentMethod.each do |payment_method|
    if [store_id].nil?
      Spree::Store.each do |store|
        payment_method.stores_ids = [store.id]
        payment_method.save
      end
    else
      payment_method.stores_ids = [store_id]
      payment_method.save
    end
  end
```

Reply:
```ruby
  Spree::PaymentMethod.each do |payment_method|
    next if payment_method.stores_ids.any?
  
    if pament_method.store_id.present?
      payment_method.stores_ids = [store_id]
    else
      payment_method.stores_ids = Spree::Store.ids
    end
    
    payment_method.save
  end
 ```

Reply:
@MatthewKennedy ☝️